src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.hotspot.aarch64/src/jdk/vm/ci/hotspot/aarch64/AArch64HotSpotRegisterConfig.java

```diff
@@ -265,11 +268,25 @@ private CallingConvention callingConvention(RegisterArray generalParameterRegist
 
             if (locations[i] == null) {
                 ValueKind<?> valueKind = valueKindFactory.getValueKind(kind);
+                int kindSize = valueKind.getPlatformKind().getSizeInBytes();
+                if (macOS && currentStackOffset % kindSize != 0) {
```

Comment:
Shouldn't this logic only be included for HotSpotCallingConventionType.NativeCall, but not JavaCall/JavaCallee?

In graal, the equivalent logic is here: https://github.com/oracle/graal/blob/cf81d2a7c3cbaa2574d0ecbbc2e76879b7719b20/substratevm/src/com.oracle.svm.core.graal.aarch64/src/com/oracle/svm/core/graal/aarch64/SubstrateAArch64RegisterConfig.java#L333